### PR TITLE
Add initial implementation of ModelLoader class for feature extractor…

### PR DESCRIPTION
**Pull request by alicjak1519**
…s and transformers uploading

It's my version of ModelLoader. I'm aware that it is not a perfect solution but I hope it's readable and user-friendly. ;)

### CODE REVIEW

File: models/feature_extractors/cnn_extrator.py [hunk 1]
```diff
@@ -1,21 +1,22 @@
 import timm
 import torch.nn as nn
 
-class Model(nn.Module):
+
+class CnnExtractor(nn.Module):
```
Comment (alicjak1519) [inline models/feature_extractors/cnn_extrator.py:5]: More descriptive than "Model" ;)

File: models/model_loader.py [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+    Only models specified in feature_extractors_ or transformers_ can be loaded.
     """
-    return model_name in defined_models()
+    feature_extractors_ = {
+        'cnn_extractor': feature_extractors.CnnExtractor
+    }
+
+    transformers_ = {
+        'vanilla_transformer': transformers.VanillaTransformer
+    }
```
Comment (alicjak1519) [inline models/model_loader.py:16]: These lines should be updated each time a new model structure is added. I'm not sure if it's the best solution.

File: models/model_loader.py [hunk 3]
```diff
@@ -1,25 +1,28 @@
-def create_model(model_name,args):
-    if is_model(model_name):   
-        package = model_name.split("/")[0]
-        name = model_name.split("/")[1]
```
Comment (alicjak1519) [inline models/model_loader.py:4]: I'm afraid it may break if the divider is diffrent (windows vs linux)

File: train.py [hunk 4]
```diff
@@ -1,12 +1,14 @@
 import argparse
-import torch
-import yaml
+
 import pytorch_lightning as pl
-from torch.utils.data import DataLoader, Dataset, random_split
```
Comment (alicjak1519) [inline train.py:5]: My mistake! I'll fix it.

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+pytorch-lightning~=1.5.6
+torch~=1.10.1
+torch-optimizer~=0.3.0
```
Comment (alicjak1519) [inline requirements.txt:3]: It could be useful :)

File: models/model_loader.py [hunk 6]
```diff
@@ -1,25 +1,28 @@
...
-    Returns:
-        (list, string): Returns list of models
-    """    
-    models = ["feature_extractors/cnn_extractor",
-            "transformers/vanilla_trasnformer",]
-    return models
\ No newline at end of file
+    def load_transformer(self, transformer_name, *args, **kwargs):
+        try:
+            return self.transformers_[transformer_name](*args, **kwargs)
```
Comment (alicjak1519) [inline models/model_loader.py:26]: Is this (*args, **kwargs) ok? Or should I do it another way?

**Review by AgaMiko — APPROVED** [review]
Everything works smoothly and well. Thanks!

File: models/model_loader.py [hunk 7]
```diff
@@ -1,25 +1,28 @@
...
+        except KeyError:
+            raise Exception(f'Transformer {transformer_name} does not exists')
```
Comment (wiktorfilipiuk) [inline models/model_loader.py:28]: niepotrzebny "s" na końcu

**Pull request merged**